src/server/zset_family.cc

```diff
@@ -130,6 +133,46 @@ OpResult<PrimeIterator> FindZEntry(const ZParams& zparams, const OpArgs& op_args
   return it;
 }
 
+// buf must be at least 12 bytes long.
+bool ToAsciiGeoHash(const std::optional<double>& val, char buf[]) {
+  if (!val)
+    return false;
+
+  double score = *val;
+
+  double xy[2];
+  GeoHashBits hash = {.bits = (uint64_t)score, .step = GEO_STEP_MAX};
+
+  if (!geohashDecodeToLongLatType(hash, xy)) {
+    return false;
+  }
+
+  /* Re-encode */
+  GeoHashRange r[2];
+  r[0].min = -180;
+  r[0].max = 180;
+  r[1].min = -90;
+  r[1].max = 90;
+
+  geohashEncode(&r[0], &r[1], xy[0], xy[1], 26, &hash);
+
+  for (int i = 0; i < 11; i++) {
+    int idx;
+    if (i == 10) {
+      /* We have just 52 bits, but the API used to output
+       * an 11 bytes geohash. For compatibility we assume
+       * zero. */
+      idx = 0;
+    } else {
+      idx = (hash.bits >> (52 - ((i + 1) * 5))) & 0x1f;
```

Comment:
Add `static_assert(strlen(GeoAlphabet) - 1 == 0x1f);`?

Reply:
Or use `% (strlen(GeoAlphabet) - 1)` instead of `& 0x1f`?

Reply:
added static assert but it's not what you think

Reply:
I'd use modulo `kGeoAlphabet.size()`, as it's clearer than a hardcoded `0x1f`, but your call. I imagine that the compiler can replace that to actually do `&` if you're worried about performance.

Reply:
oh, sorry i got brain shortcut and I thought `0x1f` is 15.